capa/helpers.py

```diff
@@ -49,12 +51,29 @@ def assert_never(value) -> NoReturn:
     assert False, f"Unhandled value: {value} ({type(value).__name__})"  # noqa: B011
 
 
+def get_format_from_report(sample: Path) -> str:
+    report = json.load(sample.open(encoding="utf-8"))
+
+    if "CAPE" in report:
+        return FORMAT_CAPE
+
+    if "target" in report and "info" in report and "behavior" in report:
+        # CAPE report that's missing the "CAPE" key,
+        # which is not going to be much use, but its correct.
+        return FORMAT_CAPE
+
+    return FORMAT_UNKNOWN
+
+
 def get_format_from_extension(sample: Path) -> str:
+    format_ = FORMAT_UNKNOWN
     if sample.name.endswith(EXTENSIONS_SHELLCODE_32):
-        return FORMAT_SC32
+        format_ = FORMAT_SC32
     elif sample.name.endswith(EXTENSIONS_SHELLCODE_64):
-        return FORMAT_SC64
-    return FORMAT_UNKNOWN
+        format_ = FORMAT_SC64
+    elif sample.name.endswith(EXTENSIONS_DYNAMIC):
+        format_ = get_format_from_report(sample)
+    return format_
 
 
 def get_auto_format(path: Path) -> str:
```

Comment:
- [ ] adjust this for CAPE format, currently leads to `DEBUG:capa.features.extractors.common:unsupported file format: 7b0a2020` or similar

Reply:
https://github.com/mandiant/capa/pull/1819